src/rgw/rgw_sal_motr.cc

```diff
@@ -3493,7 +3493,32 @@ int MotrStore::list_users(const DoutPrefixProvider* dpp, const std::string& meta
                         std::string& marker, int max_entries, void *&handle,
                         bool* truncated, std::list<std::string>& users)
 {
-    return 0;
+  int rc;
+  bool max_entires_specified = true;
+  if (max_entries <= 0) {
+    max_entries = 1000; 
+    max_entires_specified = false;
+  }
+  vector<string> keys(max_entries + 1);
+  vector<bufferlist> vals(max_entries + 1);
+
+  keys[0] = marker;
+  rc = next_query_by_name(RGW_MOTR_USERS_IDX_NAME, keys, vals);
+  if (rc < 0) {
+    ldpp_dout(dpp, 0) << "ERROR: NEXT query failed. " << rc << dendl;
+    return rc;
+  }
+  if (!(keys.back()).empty() && max_entires_specified) {
+    *truncated = true;
+    marker = keys.back(); 
+  }
+  for (int i = 0; i < int(keys.size()) - 1; i++) {
+    if (keys[i].empty()) {
```

Comment:
key contains tenant name + user name, does this shown correctly when user is listed, or its filtered based on tenant value

Reply:
> key contains tenant name + user name, does this shown correctly

Yes

It shows all the users irrespective of tenant. Filtering based on tenant is not yet supported in RGW